Documentation/compatibility/winforms-accessibility-changes-48.md

```diff
@@ -0,0 +1,105 @@
+- Changes to interaction with Narrator.
+- Changes in Accessible hierarchy. (improving navigation through UI Automation tree)
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+__How to opt in or out of these changes__
+  
+In order for the application to benefit from these changes, it must run on the .NET Framework 4.8. The application can benefit from these changes in either of the following ways:
+- It is recompiled to target the .NET Framework 4.8. These accessibility changes are enabled by default on Windows Forms applications that target the .NET Framework 4.8.
+- It targets the .NET Framework 4.7.2 or earlier version and opts out of the legacy accessibility behaviors by adding the following [AppContext Switch](https://docs.microsoft.com/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element) to the `<runtime>` section of the app config file and setting it to `false`, as the following example shows.
+
+```xml
+    <?xml version="1.0" encoding="utf-8"?>
+    <configuration>
+      <startup>
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.7"/>
+      </startup>
+      <runtime>
+        <!-- AppContextSwitchOverrides value attribute is in the form of 'key1=true|false;key2=true|false  -->
+        <AppContextSwitchOverrides value="Switch.UseLegacyAccessibilityFeatures=false;Switch.UseLegacyAccessibilityFeatures.2=false;Switch.UseLegacyAccessibilityFeatures.3=false" />
+      </runtime>
+    </configuration>
+```    
+Note that to opt in to the accessibility features added in .NET Framework 4.8, you must also opt in to accessibility features of .NET Framework 4.7.1 and 4.7.2 as well.
+Applications that target the .NET Framework 4.8 and want to preserve the legacy accessibility behavior can opt in to the use of legacy accessibility features by explicitly setting this AppContext switch to `true`.
+
+__Use of OS-defined colors in High Contrast themes__
+// To be filled.
+
+__Improved Narrator support__
+- Narrator now announces the sort direction of the <xref:System.Windows.Forms.DataGridViewColumn> when announcing an accessible name of a <xref:System.Windows.Forms.DataGridViewCell>. 
+
+__Improved Checked ListBox Accessibility support__
+- Improved Narrator support for <xref:System.Windows.Forms.CheckedListBox> control. At this moment when navigating to the <xref:System.Windows.Forms.CheckedListBox> control using keyboard, Narrator focuses the <xref:System.Windows.Forms.CheckedListBox> item and announces it.
+
+__Improved ComboBox Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.ComboBox> control with ability to use UI Automation notifications and other UI Automation features.
+
+__Improved DataGridView Accessibility support__
+- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications and other UI Automation features.
```

Comment:
```suggestion
- Enabled UI Automation support for <xref:System.Windows.Forms.DataGridView> control with ability to use UI Automation notifications, other UI Automation features, and keyboard navigation of the control.
```
Though I'm not 100% sure, please verify, I think we added f3 for sort order access

Reply:
I looked at this and found out that F3-sorting was added in 4.7.2.
.